### PR TITLE
add 4 functions for generating other objects same as the fish

### PR DESCRIPTION
**Pull request by EarthenSky**
we want to generate:
- boxes (sharp edges)
- squiggles (curves & various high & low frequency details)
- decoy fish (looks like the fish, bus surprise! It's not)

### CODE REVIEW

File: fishgen-py/src/lib.rs [hunk 1]
```diff
@@ -48,6 +48,46 @@ impl From<fishgen_rs::FishOutput> for PyFishOutput {
...
+impl PyImageOutput {
+    #[getter]
+    pub fn image<'py, 'b>(&self, m: Python<'py>) -> PyResult<&'b numpy::PyArray3<f32>>
+    where
+        'py: 'b,
+    {
+        Ok(self.image.to_pyarray(m))
+    }
+}
+
```
Comment (CatThingy) [inline fishgen-py/src/lib.rs:90]: Not entirely sure why a different struct is needed for something that holds the same data. Perhaps a global renaming of `FishOutput` to `ImageOutput` and adding an enum for the image type?

File: fishgen/src/lib.rs [hunk 2]
```diff
@@ -99,3 +102,147 @@ pub fn random_fish(
...
+    overlay(&mut image, &fish, fish_x.into(), fish_y.into());
+
+    Ok(FishOutput {
+        image: image.convert(),
+        x: fish_x,
+        y: fish_y,
+        width: fish_width,
+        height: fish_height,
+    })
+}
```
Comment (CatThingy) [inline fishgen/src/lib.rs:248]: Could the core put-image-on-white-background be separated, and have these extra functions (+ the fish one) be wrappers over that?

**Review by CatThingy — CHANGES_REQUESTED** [review]
Looks good to me, although I would like to deduplicate some code

File: fishgen-py/src/lib.rs (hunk 1 above)
Comment (EarthenSky) [reply fishgen-py/src/lib.rs:90]: My intention was just to not restructure everything, but if you're okay with it then that sounds good :)

File: fishgen/src/lib.rs (hunk 2 above)
Comment (EarthenSky) [reply fishgen/src/lib.rs:248]: this is indeed a good idea, will do

**Comment by EarthenSky** [comment]
i will make these changes whenever numpy is too slow for me (which will probably be soon)

**Pull request closed without merging**